mmseg/apis/test.py

```diff
@@ -163,3 +165,149 @@ def multi_gpu_test(model,
     else:
         results = collect_results_cpu(results, len(dataset), tmpdir)
     return results
+
+
+def progressive_single_gpu_test(model,
+                                data_loader,
+                                show=False,
+                                out_dir=None,
+                                opacity=0.5):
+    model.eval()
+    dataset = data_loader.dataset
+    num_classes = len(dataset.CLASSES)
+    prog_bar = mmcv.ProgressBar(len(dataset))
+
+    total_area_intersect = torch.zeros((num_classes, ), dtype=torch.float64)
+    total_area_union = torch.zeros((num_classes, ), dtype=torch.float64)
+    total_area_pred_label = torch.zeros((num_classes, ), dtype=torch.float64)
+    total_area_label = torch.zeros((num_classes, ), dtype=torch.float64)
+
+    cur = 0
+    for _, data in enumerate(data_loader):
+        with torch.no_grad():
+            result = model(return_loss=False, **data)
+
+        if show or out_dir:
+            img_tensor = data['img'][0]
+            img_metas = data['img_metas'][0].data[0]
+            imgs = tensor2imgs(img_tensor, **img_metas[0]['img_norm_cfg'])
+            assert len(imgs) == len(img_metas)
+
+            for img, img_meta in zip(imgs, img_metas):
+                h, w, _ = img_meta['img_shape']
+                img_show = img[:h, :w, :]
+
+                ori_h, ori_w = img_meta['ori_shape'][:-1]
+                img_show = mmcv.imresize(img_show, (ori_w, ori_h))
+
+                if out_dir:
+                    out_file = osp.join(out_dir, img_meta['ori_filename'])
+                else:
+                    out_file = None
+
+                model.module.show_result(
+                    img_show,
+                    result,
+                    palette=dataset.PALETTE,
+                    show=show,
+                    out_file=out_file,
+                    opacity=opacity)
+
+        for i in range(len(result)):
+            gt_semantic_map = dataset.get_gt_seg_map(cur + i)
+
+            area_intersect, area_union, area_pred_label, area_label = \
+                intersect_and_union(
+                    result[i], gt_semantic_map, num_classes,
+                    dataset.ignore_index, dataset.label_map,
+                    dataset.reduce_zero_label)
+
+            total_area_intersect += area_intersect
+            total_area_union += area_union
+            total_area_pred_label += area_pred_label
+            total_area_label += area_label
+
+            print(total_area_intersect / total_area_union)
+
+            prog_bar.update()
+
+            cur += len(result)
+
+    return total_area_intersect, total_area_union, total_area_pred_label, \
+        total_area_label
+
+
+# TODO: Support distributed test api
+def progressive_multi_gpu_test(model,
+                               data_loader,
+                               tmpdir=None,
+                               gpu_collect=False):
```

Comment:
May also consider GPU test to see if it is faster. 
Please also note whether it would take too much memory.